zerver/lib/zcommand.py

```diff
@@ -73,4 +77,52 @@ def change_mode_setting(
                 setting_value=False,
             )
         )
+    elif command == "digress":
+        data_keys = [
+            "old_stream",
+            "old_topic",
+            "new_stream",
+            "new_topic",
+        ]
+        if not data or not all(key in data for key in data_keys):
+            raise JsonableError(_("Invalid data."))
+
+        if data["old_stream"] == data["new_stream"] and data["old_topic"] == data["new_topic"]:
+            return dict(msg=_("Cannot digress to the same thread."))
+
+        (old_stream, ignored_old_stream_sub) = access_stream_by_name(
+            user_profile, data["old_stream"]
+        )
+
+        (new_stream, ignored_new_stream_sub) = access_stream_by_name(
+            user_profile, data["new_stream"]
+        )
+
+        old_topic_exists = user_message_exists_for_topic(
+            user_profile=user_profile,
+            recipient_id=old_stream.recipient_id,
+            topic_name=data["old_topic"],
+        )
+
+        if not old_topic_exists:
+            return dict(
+                msg=_("Old topic #**{}>{}** does not exist.").format(
+                    data["old_stream"], data["old_topic"]
+                )
+            )
+
+        old_thread_notification_string = _(
+            "{user} will be talking on the new topic: {new_location}"
```

Comment:
Let's say "{user} digressed to the new topic {new_location}`.

Reply:
Changed this string.